example/21_gemm_normalize_xdl/gemm_layernorm_xdl_fp16.cpp

```diff
@@ -0,0 +1,331 @@
+
+    auto dxs_in_element_op  = DxsInElementOp{};
+    auto dxs_out_element_op = DxsOutElementOp{M, M};
+
+    // Prepare GEMM, reduce_mean, reduce_mean_square
+    auto gemmReduce         = DeviceGemmReduceInstance{};
+    auto gemmReduce_invoker = gemmReduce.MakeInvoker();
+    auto gemmReduce_argument =
+        gemmReduce.MakeArgument(static_cast<ADataType*>(a_device_buf.GetDeviceBuffer()),
+                                static_cast<BDataType*>(b_device_buf.GetDeviceBuffer()),
+                                static_cast<CDataType*>(c_device_buf.GetDeviceBuffer()),
+                                dxs_global,
+                                M,
+                                N,
+                                K,
+                                StrideA,
+                                StrideB,
+                                StrideC,
+                                a_element_op,
+                                b_element_op,
+                                c_element_op,
+                                dxs_in_element_op,
+                                dxs_out_element_op);
+
+    if(!gemmReduce.IsSupportedArgument(gemmReduce_argument))
+    {
+        throw std::runtime_error(
+            "wrong! device_gemm with the specified compilation parameters does "
+            "not support this GEMM problem");
+    }
+
+    reduceMean_device_buf.SetZero();
+    reduceMeanSquare_device_buf.SetZero();
+
+    // Prepare LayerNorm
+    auto layerNorm             = DeviceNormalizeInstance{};
+    auto layerNorm_invoker_ptr = layerNorm.MakeInvokerPointer();
+    auto layerNorm_argument =
+        layerNorm.MakeArgumentPointer(c_device_buf.GetDeviceBuffer(),
+                                      reduceMean_device_buf.GetDeviceBuffer(),
+                                      reduceMeanSquare_device_buf.GetDeviceBuffer(),
+                                      gamma_device_buf.GetDeviceBuffer(),
+                                      beta_device_buf.GetDeviceBuffer(),
+                                      layerNorm_device_buf.GetDeviceBuffer(),
+                                      {M, N},
+                                      {StrideC, 1},
+                                      {1, 0},
+                                      {1, 0},
+                                      {0, 1},
+                                      {0, 1},
+                                      {StrideC, 1},
+                                      NormalizeFunctor{});
+
+    if(!layerNorm.IsSupportedArgument(layerNorm_argument.get()))
+    {
+        throw std::runtime_error("The runtime parameters seems not supported by the "
+                                 "Device5AryElementwise_Xdl_CShuffle instance, exiting!");
+    }
+
+    // run kernel
+    gemmReduce_invoker.Run(gemmReduce_argument, StreamConfig{nullptr, time_kernel});
+    layerNorm_invoker_ptr->Run(layerNorm_argument.get(), StreamConfig{nullptr, time_kernel});
+
```

Comment:
Need to output Performance information:
1. for GEMM+reduce: kernel time, tflops (use fma flops only, no need to calculte reduction flops), bandwidth
2. for pintwise: kernel time, and bandwidth